### DISCUSSION TITLE
Notes grouped by category

### DISCUSSION THREAD

**Discussion by DickinsonSa** [TechbeeAT/jtxBoard#387] [Ideas]
Notes seem to have little way to group when viewing other than status and classification (both marginally useful).

I have dozens of notes with various tags /categories - all of which appear in list view, and the only way to group at the moment is by filtering them individually (time consuming). 

Categories (tags) seem to be a more universally recognised way of grouping (other than notebooks) and yet both these methods are under-utilised for grouping.

Can I put forward for discussion/enhancement please the inclusion of  category in the grouping option for notes?

This would include a 'no category assigned' group also.

**Comment by patrickunterwegs** [comment]
Hey @DickinsonSa , 
sure, it absolutely makes sense. 

For a grouping-option for collections and accounts I have created #388 , this should be an easy one.

Group by category is a bit more tricky. The problem is, that currently all group options work on one base list that is grouped by an attribute that can appear only once. When grouping by category, it needs to be considered that an entry can be in multiple categories, which also means that the entry can appear in multpile groups. For 2.4.0 I have already done a structural change that will make it easier to achieve that. 

But it will need a bit of time. I'm currently on vacation and need to slow down a bit 😉

**Comment by rogercreagh** [comment]
In my world I think of category and tag as two different things - we are grappling here with a deficiency of the VJOURNAL format which uses the word Category for what I think of as Tags.

In most applications an item can belong to only one category, which may be arranged in a hierarchy of categories, but an item can only be one type (or class, or category) of thing. Members of the category may inherit attributes from its parent type/class/category. Think of biological classification schemes, or library systems like the DeweyDecimal classification, or OOP languages.

Clearly you can use categories in this sense for grouping.

What we have in VJOURNAL is something called Categories, but where an item can belong to multiple categories. For me, and most systems I am aware of, this is a Tag, not a Category. Something can have multiple tags (labels) attached to it. In many systems there is no hierarchy for tags, some do allow a hierarchy which provides a convenient way of grouping tags together under a particular type. 

There is almost no sensible way of grouping and showing an entire set by tags if items in the set can have multiple tags -  you have to have the items appearing multiple times, which to me is neither sensible or useful.

What you can do usefully, and very effectively, is filter the entire set by a tag or combination of tags. jtxBoard is partly there with this - you can filter by tags, but with multiple tags only the combination of ANY of the selected tags. Ideally I would want to be able to select to either show items with ANY or ALL or NONE of the selected tags (eg by using another radio button for the logic if multiple tags were selected).

VJOURNAL does use what I would call a category (only one can be assigned) for Status and Classification, but sadly we only have fixed values - otherwise you could add your own status values for whatever categories (not tags misnamed as categories) you wanted to use and easily group by them. 

I think we discussed elsewhere that either status or classification (I can't remember which) does allow new values to be defined, but the feeling was this might cause problems with other apps not recognising them.

It really needs a 'big' organisation to engage with the standards RFC process and push for getting proper categories for VJOURNAL.

**Comment by rogercreagh** [comment]
I think what I am saying above is that I think that grouping by Category-Tags is not a good idea, and potentially confusing.

**Comment by patrickunterwegs** [comment]
Hmmm, well, I basically agree with you that the word category is misleading and should better be called tag in the standard. 
But what was your original idea for the grouped notes by category?
It is possible to implement, it's just the fact that one entry could appear multiple times that might be confusing. On the other hand the user decided to give multiple categories...

Btw. I'm still thinking of adding an option to define individual status options. I just want to do some testing before to make sure common servers can deal with it.

**Comment by DickinsonSa** [comment]
Hi Patrick,

You really should relax if you're on your break - it will be here when you get back. But if you insist on working ...

Ok, so I've had a rethink in light of some comments made and realised that I hadn't given enough consideration to the standards as well as the need of other people.

So my thoughts basically are these - rename categories to tags. Keep them as they are (multi selectable and also filterable) - so no change there other than just categories becomes tags - just the wording changes.

Introduce a new category called notebook. The user can enter as many names as they like (like categories /tags are now. Each note can be assigned to none or one (ONLY ONE) of the notebooks. These may or may not be filterable - you could advise on best approach to filters. I guess they could filter by notebook if they wished.

However, the notebook category IS GROUPABLE - ie all notes would appear under their respective notebook - nice and clean and no duplicates anywhere!

Does this make any sense to you?

I haven't been through the VTODO standards but I don't think it departs from what you already have - it's just laying it out slightly differently. But it's also laying down firmly the notion of notes and notebooks - something other apps do - though jtx board is far superior to any other apps currently. It's just that people relate to tags/notes /notebooks.

As I said, I haven't been through the fine details yet, but I think it keeps to the standards that are already in place and in use in the jtx so far.

What do you think?

Ps happy to support whatever decision. I have almost finished  degoogling now using the app - many thanks.

Regards
Dave

21 Mar 2023 14:24:13 Patrick Lang ***@***.***>:
